### PR TITLE
Feat/performance fixes

### PR DESCRIPTION
**Pull request by hipdev**
This pull request (PR) outlines the concept of why deploying a static website is typically faster. It includes the following changes:

- Eliminates 'noStore' and caching mechanisms to prevent dynamic rendering or partial pre-rendering.
- Removes certain animations to align with Google's guidelines.
- Addresses minor style issues to avoid penalties from Google.

Please note that this is still a conceptual overview and further improvements are required in the code. The logic related to posts has not been extensively modified.

### CODE REVIEW

File: postcss.config.js [hunk 1]
```diff
@@ -2,6 +2,5 @@ module.exports = {
   plugins: {
     tailwindcss: {},
     autoprefixer: {},
-    ...(process.env.NODE_ENV === 'production' ? { cssnano: {} } : {}),
```
Comment (jahirfiquitiva) [inline postcss.config.js:5]: please keep this

File: package.json [hunk 2]
```diff
@@ -37,7 +36,6 @@
     "@typescript-eslint/parser": "^7.2.0",
     "autoprefixer": "^10.4.18",
     "colord": "^2.9.3",
-    "cssnano": "^6.1.0",
```
Comment (jahirfiquitiva) [inline package.json:40]: please keep this

File: package.json [hunk 3]
```diff
@@ -1,5 +1,4 @@
 {
-  "private": true,
```
Comment (jahirfiquitiva) [inline package.json:2]: please keep this

File: .env [hunk 4]
```diff
@@ -1,2 +1,2 @@
-IS_TEMPLATE=true
```
Comment (jahirfiquitiva) [inline .env:1]: do not change this please (or do not commit the changes)

File: src/components/atoms/main.tsx [hunk 5]
```diff
@@ -14,5 +14,5 @@ export const Main = tw.main.attrs({ style: { maxWidth: MAX_SITE_WIDTH } })`
   mx-auto
   tablet-md:pt-32
   tablet-md:px-0
-  motion-safe:animate-fade-in
```
Comment (jahirfiquitiva) [inline src/components/atoms/main.tsx:17]: would you mind finding all occurrences of `motion-safe:animate-fade-in` and also remove the tailwind config for it?

File: src/components/ui/home/index.tsx [hunk 6]
```diff
@@ -23,7 +23,7 @@ export const Intro = () => {
           alt={'Photo of Jahir Fiquitiva'}
           width={160}
           height={160}
-          quality={100}
+          quality={60}
```
Comment (jahirfiquitiva) [inline src/components/ui/home/index.tsx:26]: Let's use the default (75%)
```suggestion
```

File: src/components/ui/home/index.tsx [hunk 7]
```diff
@@ -50,7 +50,7 @@ export const Intro = () => {
         </h1>
 
         <p className={'text-pretty flex flex-col gap-2'}>
-          <span style={{ maxWidth: '42ch' }}>
+          <span className={'md:max-w-[26rem]'}>
```
Comment (jahirfiquitiva) [inline src/components/ui/home/index.tsx:53]: Please double check these values, they are meant to leave a "safe" empty space around the photo.

Additionally, let's not use a breakpoint, `max-w` will be safe in mobile devices
```suggestion
          <span className={'max-w-[26rem]'}>
```

Shot 2024-03-21 at 16 04 40@2x

**Review by jahirfiquitiva — CHANGES_REQUESTED** [review]
Interesting changes and approach. Thank you so much for your contribution! 🫶

File: src/components/ui/home/index.tsx (hunk 7 above)
Comment (jahirfiquitiva) [reply src/components/ui/home/index.tsx:53]: Oh, and please try using a value from the spacing configured in tailwind, rather than an arbitrary value

File: README.md [hunk 8]
```diff
@@ -1,4 +1,4 @@
-# [jahir.dev](https://jahir.dev)
+# [jahir.dev](https://jahir-site.vercel.app/)
```
Comment (jahirfiquitiva) [inline README.md:1]: please do not change this file

File: package.json [hunk 9]
```diff
@@ -15,6 +15,7 @@
   },
   "dependencies": {
     "canvas-confetti": "^1.9.2",
+    "cssnano": "^6.1.1",
```
Comment (jahirfiquitiva) [inline package.json:18]: this should be a dev dependency

File: src/actions/stars.ts [hunk 10]
```diff
@@ -16,7 +14,6 @@ export const getStars = async (
   repo: string,
   owner: string = 'jahirfiquitiva',
 ): Promise<string | null> => {
-  noStore();
```
Comment (jahirfiquitiva) [inline src/actions/stars.ts:19]: I think removing `noStore` is causing the counters to not be properly updated

**Review by jahirfiquitiva — CHANGES_REQUESTED** [review]

File: package.json [hunk 11]
```diff
@@ -23,31 +23,31 @@
     "swr": "^2.2.5"
   },
   "devDependencies": {
-    "@commitlint/cli": "19.1.0",
+    "@commitlint/cli": "19.2.1",
```
Comment (jahirfiquitiva) [inline package.json:26]: does upgrading dependencies have any improvements in the website performance?

otherwise please revert these changes and keep the pr as simple as possible

**Review by jahirfiquitiva — CHANGES_REQUESTED** [review]

File: package.json (hunk 11 above)
Comment (hipdev) [reply package.json:26]: Is good to have them updated, but I can revert it.

File: src/app/layout.tsx [hunk 12]
```diff
@@ -55,8 +55,7 @@ export default function RootLayout(props: PropsWithChildren) {
       <head>
         <Meta />
         <Script
-          async
-          defer
+          strategy={'afterInteractive'}
```
Comment (jahirfiquitiva) [inline src/app/layout.tsx:58]: please revert this change, it is already the default strategy
https://nextjs.org/docs/app/api-reference/components/script#strategy

**Review by jahirfiquitiva — CHANGES_REQUESTED** [review]

File: src/app/layout.tsx (hunk 12 above)
Comment (hipdev) [reply src/app/layout.tsx:58]: You're right!

File: src/app/layout.tsx [hunk 13]
```diff
@@ -55,11 +55,10 @@ export default function RootLayout(props: PropsWithChildren) {
       <head>
         <Meta />
         <Script
-          async
-          defer
           src={'https://umami.jahir.dev/script.js'}
           data-website-id={umamiWebsiteId}
           data-domains={'jahir.dev'}
+          lazyOnLoad
```
Comment (jahirfiquitiva) [inline src/app/layout.tsx:61]: this prop does not exist, it should be
```suggestion
          strategy={'lazyOnLoad'}
```

**Review by jahirfiquitiva — CHANGES_REQUESTED** [review]

File: src/app/layout.tsx [hunk 14]
```diff
@@ -55,11 +55,10 @@ export default function RootLayout(props: PropsWithChildren) {
       <head>
         <Meta />
         <Script
-          async
-          defer
           src={'https://umami.jahir.dev/script.js'}
           data-website-id={umamiWebsiteId}
           data-domains={'jahir.dev'}
+          strategy={'lazyOnLoad'}
```
Comment (jahirfiquitiva) [inline src/app/layout.tsx:61]: This has a typo 😅 
```suggestion
          strategy={'lazyOnload'}
```

**Review by jahirfiquitiva — CHANGES_REQUESTED** [review]

**Review by jahirfiquitiva — APPROVED** [review]
